content/en/guide/v10/forms.md

```diff
@@ -117,61 +176,241 @@ class MySelect extends Component {
     this.setState({ value: e.currentTarget.value });
   }
 
-  onSubmit = e => {
-    alert("Submitted " + this.state.value);
-    e.preventDefault();
-  }
-
   render(_, { value }) {
     return (
-      <form onSubmit={this.onSubmit}>
-        <select value={value} onChange={this.onChange}>
+      <div class="form-example">
+        <select onChange={this.onChange}>
           <option value="A">A</option>
           <option value="B">B</option>
           <option value="C">C</option>
         </select>
-        <button type="submit">Submit</button>
-      </form>
+        <p>You selected: {value}</p>
+      </div>
     );
   }
 }
 // --repl-after
 render(<MySelect />, document.getElementById("app"));
 ```
 
-## Checkboxes & Radio Buttons
+```jsx
+// --repl
+import { render } from "preact";
+import { useState } from "preact/hooks";
+// --repl-before
+function MySelect() {
+  const [value, setValue] = useState('');
+
+  return (
+    <div class="form-example">
+      <select onChange={(e) => setValue(e.currentTarget.value)}>
+        <option value="A">A</option>
+        <option value="B">B</option>
+        <option value="C">C</option>
+      </select>
+      <p>You selected: {value}</p>
+    </form>
+  );
+}
+// --repl-after
+render(<MySelect />, document.getElementById("app"));
+```
+
+</tab-group>
 
-Checkboxes and radio buttons (`<input type="checkbox|radio">`) can initially cause confusion when building controlled forms. This is because in an uncontrolled environment, we would typically allow the browser to "toggle" or "check" a checkbox or radio button for us, listening for a change event and reacting to the new value.  However, this technique does not transition well into a world view where the UI is always updated automatically in response to state and prop changes.
+## Basic Forms
 
-> **Walk-Through:** Say we listen for a "change" event on a checkbox, which is fired when the checkbox is checked or unchecked by the user.  In our change event handler, we set a value in `state` to the new value received from the checkbox.  Doing so will trigger a re-render of our component, which will re-assign the value of the checkbox to the value from state.  This is unnecessary, because we just asked the DOM for a value but then told it to render again with whatever value we wanted.
```

Comment:
https://preactjs.com/guide/v10/forms#checkboxes--radio-buttons

I've read this through a dozen or so times and I'm still not quite sure what it is suggesting here. Using the updated state value from `onChange` & rerendering based upon it is silly & wasteful, so instead, use the opposite of the current value (as it's just a toggle) and rerender with that instead? It's the same issue, just a different format (and one I'd argue is more problematic, as it allows component state to get out-of-sync with the DOM in some cases).

Not sure if that makes sense to anyone else, very, very possible I'm misunderstanding entirely, but I stripped this comment out as it seemingly makes little sense to me at the moment. I don't understand what it's advocating for or why that's an improvement.